geps/gep-1619.md

```diff
@@ -157,49 +281,401 @@ Generally, the implementation API programs the dataplane API; however these two
+- Name=_value_
 - Expires=_date_
 - Max-Age=_number_
 - Domain=_domain_
 - Path=_path-value_
 - Secure
 - HttpOnly
 
-However, there are de-facto standard attributes that have been proposed as updates to [rfc6265](https://www.rfc-editor.org/rfc/rfc6265#section-5.2):
-- SameSite=_[Strict|Lax|None]_ ([Draft RFC Update](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-cookie-same-site-00))
-- Partitioned ([Draft RFC Update](https://www.ietf.org/archive/id/draft-cutler-httpbis-partitioned-cookies-01.html))
-
-**Session Cookie Strategy**
+Other cookie attributes not defined by RFC6265, but are captured in draft RFCs and could be considered de facto
+standards due to wide acceptance are:
+- SameSite=[Strict|Lax|None]
+- Partitioned
 
-Session cookie strategy is configuration specific to the creation of the cookie by the dataplane. These are not specific configuration parameters of the cookie itself, but instead the operation to take when inserting the cookie such as:
-- Rewriting an existing cookie value
-- Inserting a new cookie
-- Prefixing an existing cookie
-- Preserving an existing cookie value
-- Only adding cookies on specific HTTP methods (POST/GET)
+All cookie attributes are considered as opaque values in this spec and are to be determined by the individual
+implementations. However, certain cookie attributes require special consideration, which we will discuss in detail.
+
+##### Path
+
+The `Path` attribute should be automatically determined by implementors. When the `SessionPersistencePolicy` policy is
+attached to an xRoute, the implementor should interpret the path as configured on the xRoute. However, if the
+`SessionPersistencePolicy` policy is attached to a service, the path should be left unset.
+
+To interpret the `Path` attribute from an xRoute, implementors should take note of the following:
+1. For an xRoute that matches all paths, the `Path` should be set to `/`.
+2. For an xRoute that has multiple paths, the `Path` should be interpreted based on the path taken.
+3. For an xRoute using a path that is a regex, the `Path` should be set to the longest non-regex prefix (.e.g. if the
+   path is /p1/p2/*/p3 and the request path was /p1/p2/foo/p3, then the cookie path would be /p1/p2).
+
+##### Secure, HttpOnly, SameSite
+
+The `Secure`, `HttpOnly`, and `SameSite` cookie attributes are security-related. The API implementers should follow the
+security-by-default principle and configure these attributes accordingly. This means enabling `Secure` and `HttpOnly`,
+and setting `SameSite` to `Strict`.
+
+#### Session Persistence API with GAMMA
+
+The object of the [GAMMA (Gateway API for Mesh Management and Administration)](https://gateway-api.sigs.k8s.io/contributing/gamma/)
+initiative is to provide support for service mesh and mesh-adjacent use-cases with Gateway API. Mesh traffic routing with
+Mesh traffic routing with Gateway API is accomplished with `xRoutes` specifying service-to-service traffic opposed to
```

Comment:
Why ? If a service is using service attachment for Ingress, it would not make sense to use route attachment for internal traffic. I think GAMMA should follow the same rules as Gateway ( in particular since it's very common for 'internal' gateways to be used in mesh)

Reply:
Didn't quite understand this. Can you elaborate?

Reply:
For context, this line isn't saying anything about session persistence. I was trying to describe GAMMA, but it's likely I need a more accurate definition.

But I think I understand what costin is saying. This is definition is inaccurate and needlessly specific. I updated with content from https://gateway-api.sigs.k8s.io/concepts/gamma/#the-gamma-initiative-gateway-api-for-service-mesh to be more accurate.

Reply:
Thanks.